cmd/crank/install.go

```diff
@@ -141,8 +148,22 @@ func (c *installProviderCmd) Run(k *kong.Context) error {
 	kube := typedclient.NewForConfigOrDie(ctrl.GetConfigOrDie())
 	res, err := kube.Providers().Create(context.Background(), cr, metav1.CreateOptions{})
 	if err != nil {
+		if msg := warnIfNotFound(err); msg != "" {
+			fmt.Fprintf(k.Stdout, warnPluginOutdated, msg, version.New().GetVersionString())
+		}
 		return errors.Wrap(err, "cannot create provider")
```

Comment:
@khos2ow how about just passing `warnIfNotFound(err)` here and passing in the alternate message?
```suggestion
		return errors.Wrap(err, warnIfNotFound(err, "cannot create provider"))
```

Reply:
I originally tried similar approach (but with `errors.WithMessage()`) and I personally didn't like the output message. It reads as:

```
kubectl crossplane: error: cannot create provider: kubectl-crossplane plugin might be out of date: the server could not find the requested resource (post providers.pkg.crossplane.io)
```

The order is wrong, IMO. `kubectl crossplane: error: <title>: <some speculation>: <what actually went wrong>`. I mean it's kinda ok too. What do you think?

If you like the output error message then I'd actually convert your suggestion to this:

```suggestion
		return errors.Wrap(warnIfNotFound(err), "cannot create provider")
```

Reply:
hmm yeah it is kind of a peculiar looking error. However, I do think it is better than printing the `kubectl-crossplane plugin might be out of date` on a line before the error message. The heavy-handed solution would be to implement a custom wrapper, but I would be fine to what your change here in the near term.

Reply:
I wanted to show the warning after the error (which wasn't possible due to current implementation of logging) and settled for the next thing (to show it before). But now to think of it, I prefer a one liner better. Will update.